pydra/engine/workers.py

```diff
@@ -1,901 +1,1043 @@
+            else:  # it could be tuple that includes pickle files with tasks and inputs
+                ind, task_main_pkl, task_orig = runnable
+                future = client.submit(load_and_run, task_main_pkl, ind, rerun)
+                result = await future
+        return result
+
+    def close(self):
+        """Finalize the internal pool of tasks."""
+        pass
+
+
+class PsijWorker(Worker):
+    def __init__(self, subtype, **kwargs):
+        """
+        Initialize PsijWorker.
+
+        Parameters
+        ----------
+        subtype : str
+            Scheduler for PSI/J.
+        """
+        try:
+            import psij
+        except ImportError:
+            logger.critical("Please install psij.")
+            raise
+        logger.debug("Initialize PsijWorker")
+        self.psij = psij
+        self.subtype = subtype
+
+    def run_el(self, interface, rerun=False, **kwargs):
+        """Run a task."""
+        return self.exec_psij(interface, rerun=rerun)
+
+    def make_spec(self, cmd=None, arg=None):
+        """
+        Create a PSI/J job specification.
+
+        Parameters
+        ----------
+        cmd : str, optional
+            Executable command. Defaults to None.
+        arg : list, optional
+            List of arguments. Defaults to None.
+
+        Returns
+        -------
+        psij.JobSpec
+            PSI/J job specification.
+        """
+        spec = self.psij.JobSpec()
+        spec.executable = cmd
+        spec.arguments = arg
+
+        return spec
+
+    def make_job(self, spec, attributes):
+        """
+        Create a PSI/J job.
+
+        Parameters
+        ----------
+        spec : psij.JobSpec
+            PSI/J job specification.
+        attributes : any
+            Job attributes.
+
+        Returns
+        -------
+        psij.Job
+            PSI/J job.
+        """
+        job = self.psij.Job()
+        job.spec = spec
+        return job
+
+    async def exec_psij(self, runnable, rerun=False):
+        """
+        Run a task (coroutine wrapper).
+
+        Raises
+        ------
+        Exception
+            If stderr is not empty.
+
+        Returns
+        -------
+        None
+        """
+        import pickle
+        import os
+
+        jex = self.psij.JobExecutor.get_instance(self.subtype)
+        absolute_path = os.path.dirname(__file__)
+
+        if isinstance(runnable, TaskBase):
+            cache_dir = runnable.cache_dir
+            file_path = os.path.join(cache_dir, "my_function.pkl")
+            with open(file_path, "wb") as file:
+                pickle.dump(runnable._run, file)
+            func_path = os.path.join(absolute_path, "run_pickled.py")
+            spec = self.make_spec("python", [func_path, file_path])
+        else:  # it could be tuple that includes pickle files with tasks and inputs
+            cache_dir = runnable[-1].cache_dir
+            file_path_1 = os.path.join(cache_dir, "my_function.pkl")

```

Comment:
it's ok to use `os.path`, but we mostly use `Path` from `pathlib` and build new path [this way](https://github.com/nipype/pydra/blob/0245cdcf940abcdf58443f754e958a0b1bcb02c5/pydra/engine/core.py#L414)

Reply:
How can I get the absolute location of `run_pickled.py` using this?

Reply:
check https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve

Reply:
Done